aspnetcore/web-api/jsonpatch.md

```diff
@@ -0,0 +1,221 @@
+
+[!code-json[](jsonpatch/samples/2.2/JSON/add.json)]
+
+## The remove operation
+
+* If `path` points to an array element: removes the element.
+* If `path` points to a property:
+  * If resource to patch is a dynamic object: removes the property.
+  * If resource to patch is a static object: 
+    * If the property is nullable: sets it to null.
+    * If the property is non-nullable, sets it to `default<T>`.
+
+The following sample patch document sets `CustomerName` to null and deletes `Orders[0]`.
+
+[!code-json[](jsonpatch/samples/2.2/JSON/remove.json)]
+
+## The replace operation
+
+This operation is functionally the same as a `remove` followed by an `add`.
+
+The following sample patch document sets the value of `CustomerName` and replaces `Orders[0]`with a new `Order` object.
+
+[!code-json[](jsonpatch/samples/2.2/JSON/replace.json)]
+
+## The move operation
+
+* If `path` points to an array element: copies `from` element to location of `path` element, then runs a `remove` operation on the `from` element.
+* If `path` points to a property: copies value of `from` property to `path` property, then runs a `remove` operation on the `from` property.
+* If `path` points to a nonexistent property:
+  * If the resource to patch is a static object: the request fails.
+  * If the resource to patch is a dynamic object: copies `from` property to location indicated by `path`, then runs a `remove` operation on the `from` property.
+
+The following sample patch document:
+
+* Copies the value of `Orders[0].OrderName` to `CustomerName`.
+* Sets `Orders[0].OrderName` to null.
+* Moves `Orders[1]` to before `Orders[0]`.
+
+[!code-json[](jsonpatch/samples/2.2/JSON/move.json)]
+
+## The copy operation
+
+This operation is functionally the same as a `move` operation without the final `remove` step.
+
+The following sample patch document:
+
+* Copies the value of `Orders[0].OrderName` to `CustomerName`.
+* Inserts a copy of `Orders[1]` before `Orders[0]`.
+
+[!code-json[](jsonpatch/samples/2.2/JSON/copy.json)]
+
+## The test operation
+
+If the value at the location indicated by `path` is different from the value provided in `value`, the request fails. In that case, the whole PATCH request fails even if all other operations in the patch document would otherwise succeed.
+
+The `test` operation is commonly used to prevent an update when there's a concurrency conflict.
+
+The following sample patch document has no effect if the initial value of `CustomerName` is "John", because the test fails:
+
+[!code-json[](jsonpatch/samples/2.2/JSON/test-fail.json)]
+
+## Get the code
+
+[View or download sample code](https://github.com/aspnet/Docs/tree/master/aspnetcore/web-api/jsonpatch/samples/2.2). ([How to download](xref:index#how-to-download-a-sample)).
+
+To try out the sample app, run the project, and then send HTTP requests with the following settings:
```

Comment:
To test the sample, run the app and send HTTP requests with the following settings: